versionedRouter.js

```diff
@@ -412,34 +412,60 @@ async function handleValidation(ctx) {
   });
 }
 
+async function isValidRouterDest(event, destType) {
+  const isCdkV2Dest = isCdkV2Destination(event);
+  if (isCdkV2Dest) {
+    try {
+      await getWorkflowEngine(destType, TRANSFORMER_METRIC.ERROR_AT.RT);
+      return true;
+    } catch (error) {
+      return false;
+    }
+  }
+  try {
+    const routerDestHandler = getDestHandler("v0", destType);
+    return routerDestHandler?.processRouterDest !== undefined;
+  } catch (error) {
+    return false;
+  }
+}
 async function routerHandleDest(ctx) {
   const respEvents = [];
   let destType;
   try {
     const { input } = ctx.request.body;
     destType = ctx.request.body.destType;
+    if (!Array.isArray(input)) {
+      ctx.status = 400;
+      ctx.body = "input should be an array";
+      return null;
+    }
```

Comment:
Let's revert this change for now, as it doesn't align with the existing non-CDK flow.

Reply:
Sure